### PR TITLE
Rework TPM workflow

### PR DESCRIPTION
**Pull request by Itxaka**
- Allows to pass tpm pcr and public key pcr banks to bind to
 - Lots of more logging
 - Separate luksify and luksify with measurements to make it clearer
 - Drop the luks version, its not used anywhere and luks2 is safer
 - log the actual command being run on debug
 - enable debug mode for cryptenroll
 - print debug content from cryptenroll on failure in order to make easier to troubleshoot
 - As a side effect, it also avoids the print to stdout when adding a tpm key that it appears when used as a library
 - even more logging

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -25,27 +26,39 @@ func main() {
...
-						Usage: "luks version to use",
-					},
 					&cli.BoolFlag{
 						Name:  "tpm",
-						Usage: "Use TPM to lock the partition",
+						Usage: "Use TPM measurements to lock the partition",
+					},
+					&cli.StringSliceFlag{
+						Name:  "tpm-pcrs",
+						Usage: "tpm pcrs to bind to (single measurement)",
```
Comment (jimmykarily) [inline main.go:37]:
```suggestion
						Usage: "tpm pcrs to bind to (single measurement) . Only applies when --tpm is also set.",
```

File: main.go [hunk 2]
```diff
@@ -25,27 +26,39 @@ func main() {
...
+					},
+					&cli.StringSliceFlag{
+						Name:  "public-key-pcrs",
+						Usage: "public key pcrs to bind to (policy)",
```
Comment (jimmykarily) [inline main.go:41]:
```suggestion
						Usage: "public key pcrs to bind to (policy). Only applies when --tpm is also set.",
```

**Review by jimmykarily — APPROVED** [review]
2 minor things. Good job!

File: pkg/lib/lock.go [hunk 3]
```diff
@@ -48,87 +48,150 @@ func getRandomString(length int) string {
...
+	args := []string{
+		"--tpm2-public-key=/run/systemd/tpm2-pcr-public-key.pem",
+		fmt.Sprintf("--tpm2-public-key-pcrs=%s", strings.Join(publicKeyPcrs, "+")),
+		fmt.Sprintf("--tpm2-pcrs=%s", strings.Join(pcrs, "+")),
+		"--tpm2-signature=/run/systemd/tpm2-pcr-signature.json",
+		"--tpm2-device-key=/run/systemd/tpm2-srk-public-key.tpm2b_public",
+		part}
+	logger.Debug().Str("args", strings.Join(args, " ")).Msg("running command")
+	cmd := exec.Command("systemd-cryptenroll", args...)
+	cmd.Env = append(cmd.Env, fmt.Sprintf("PASSWORD=%s", pass), "SYSTEMD_LOG_LEVEL=debug") // cannot pass it via stdin
```
[the remark this answers is not in the document]
Comment (Itxaka) [reply pkg/lib/lock.go:139]: Yep, its the only way we can print the debug if something fails.

Should not affect anything as we are only checking for the exit status, which should be zero. But its immensely useful for when we find any problems :)

**Review by mauromorales — APPROVED** [review]

**Pull request merged**